core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectOrchestrator.scala

```diff
@@ -198,7 +197,21 @@ class BillingProjectOrchestrator(ctx: RawlsRequestContext,
             )
           )
       }
-      jobControlId <- projectLifecycle.initiateDelete(projectName, ctx)
+      billingProfileId <- billingRepository.getBillingProfileId(projectName)
+
+      maybeGoogleProject = billingProfileId match {
+        case Some(profileId) =>
+          val billingProfile = billingProfileManagerDAO.getBillingProfile(UUID.fromString(profileId), ctx)
+          val googleOrUnknownPlatform = billingProfile match {
+            case Some(profile) => profile.getCloudPlatform.equals(CloudPlatform.GCP)
+            case None          => true
+          }
+          googleOrUnknownPlatform
+        case None => true
+      }
+      _ = googleBillingProjectLifecycle.maybeCleanupResources(projectName, maybeGoogleProject, ctx)
+
+      jobControlId <- azureBillingProjectLifecycle.maybeCleanupResources(projectName, maybeGoogleProject, ctx)
```

Comment:
This will attempt to delete the landing zone if the billing project record contains a `landingZoneId`. I left this in `AzureBillingProjectLifecycle` because it is an Azure-specific resource.